app/models/miq_worker.rb

```diff
@@ -462,6 +462,21 @@ def clean_active_messages
       # and each message can only have 1 task, so this should be very fast even for many workers.
       m.miq_task.update_status(MiqTask::STATE_FINISHED, MiqTask::STATUS_ERROR, message)
     end
+
+    processed_messages.where.not(:tracking_label => nil).each do |m|
+      # TODO: This could be moved down into the MiqQueue model too.  MiqRequest and MiqRequesTask and subclasses inject their logic into the queue
+      # with their tracking_label format so this logic doesn't really belong here but possibly not even MiqQueue.  Find a good home for this.
+      # TODO Add tests for wherever this lives.
+      request_id, _class_name, request_or_task_id = m.parse_tracking_label
+
+      if request_id == request_or_task_id
+        # for a request, we abort it all of it's tasks via the queue
+        MiqRequest.find(request_id).abort_with_message(message)
+      else
+        # for a request task, abort it and it's parent request
+        MiqRequestTask.find(request_or_task_id).abort_with_message(message)
```

Comment:
I think I'm going to try to move this logic into a mixin that both request and request task can pull in.  They're the ones that specify a tracking label so they should have the logic for finding queue messages of their type:

```
joerafaniello@Joes-MBP-2 manageiq % rg -M 300 -A 15 -e "def tracking_label_id" ./ `gem env home`
./app/models/miq_request_task/state_machine.rb
6:  def tracking_label_id
7-    "r#{miq_request_id}_#{self.class.base_model.name.underscore}_#{id}"
8-  end

./app/models/miq_request.rb
179:  def tracking_label_id
180-    "r#{id}_#{self.class.name.underscore}_#{id}"
181-  end
182-
```